Add info about needed privileges for constraint commands (#1174)
neo4j/neo4j-documentation: cypher/cypher-docs/src/docs/dev/ql/constraints/syntax.adoc

The create constraint command is optionally idempotent, with the default behavior to throw an error if you attempt to create the same constraint twice.
With the `IF NOT EXISTS` flag, no error is thrown and nothing happens should a constraint with the same name or same schema and constraint type already exist.
It may still throw an error if conflicting data, indexes, or constraints exist.

> Examples of this would be nodes with missing properties, indexes with the same name, or constraints with same schema but a different constraint type.

Reviewer: ```suggestion
Examples of this are nodes with missing properties, indexes with the same name, or constraints with same schema but a different constraint type.
```

== Syntax for listing constraints

List constraints in the database, either all or filtered on constraint type.
This requires <<administration-security-administration-database-constraints, the `SHOW CONSTRAINT` privilege>>.

> The simple version of the command allows for a where clause and will give back the default set of output columns:

Reviewer: ```suggestion
The simple version of the command allows for a `WHERE` clause and will give back the default set of output columns:
```
While in this section, may as well fix this, if it's straightforward.
